### PR TITLE
Update credential requirements

### LINKED ISSUE #11: Failure when ClientSecurityContext used without credentials
ClientSecurityContext's `credential` parameter is optional. The docstring says that it defaults to the current user context if not set. However, when I don't pass it I get an exception:
```python
import sspilib
ctx = sspilib.ClientSecurityContext('s/h')
ctx.step(None)
```
generates
```
OSError: [WinError -2146893055] The handle specified is invalid
```
If I pass default UserCredential, step works fine:
```python
ctx = sspilib.ClientSecurityContext('s/h', credential=sspilib.UserCredential())
ctx.step(None)
```

### PR DESCRIPTION
**Pull request by jborean93**
Update the documentation that specifies a credential MUST be specified when creating a security context. This is technically a breaking change but it should be a minimal one.

I've done a search on github for any users of this library and could not find any situations where the change in positional arguments would cause any issues. That doesn't rule it out completely but I'm fairly confident this should be a minimally impacted change. To have code that works with both then using the keyword in the constructor will work with both for example:

```python
sspilib.ClientSecurityContext(credential=cred)
```

Fixes: https://github.com/jborean93/sspilib/issues/11

### CODE REVIEW

File: src/sspilib/_sec_context.py [hunk 1]
```diff
@@ -340,18 +340,18 @@ class ClientSecurityContext(SecurityContext):
     context has been negotiated and is complete.
 
     Args:
-        target_name: The target service's name, typically the SPN.
-        flags: Custom ISC REQ flags to use.
         credential: A credential to use for authentication, defaults to the
             current user context if not set.
```
Comment (eltoder) [inline src/sspilib/_sec_context.py]: There is no defaulting anymore.

Comment (jborean93) [reply src/sspilib/_sec_context.py]: Thanks for the pickup, has been updated.

File: src/sspilib/_sec_context.py [hunk 2]
```diff
@@ -329,29 +333,24 @@ class ClientSecurityContext(SecurityContext):
...
         target_name: The target service's name, typically the SPN.
         flags: Custom ISC REQ flags to use.
-        credential: A credential to use for authentication, defaults to the
-            current user context if not set.
         channel_bindings: Optional channel bindings to tie the context to.
     """
 
     def __init__(
         self,
+        credential: raw.CredHandle,
```
Comment (eltoder) [inline src/sspilib/_sec_context.py:351]: Sorry, I did not catch this earlier, but placing `credential` after `target_name` is probably a less disruptive change. `target_name` was a single required argument, so in my code I had `ClientSecurityContext(spn, credential=UserCredential())`, which will be broken by this change. I now updated to `ClientSecurityContext(target_name=spn, credential=UserCredential())`, but making credential the second argument would work without any changes.

Comment (jborean93) [reply src/sspilib/_sec_context.py:351]: Yea I did consider this and found that having it first aligns this with the server context object and as it was already a breaking change and I couldn't find any active examples that used the failing way in GitHub search I decided to just go with it.

Comment (eltoder) [reply src/sspilib/_sec_context.py:351]: I found one example in pyinstaller's tests: https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/5a553ff210635443d41ed5dab97f47196fcfa8e3/src/_pyinstaller_hooks_contrib/tests/test_libraries.py#L1770

Comment (eltoder) [reply src/sspilib/_sec_context.py:351]: Which is actually the only use I've found outside of this repo :)

Comment (jborean93) [reply src/sspilib/_sec_context.py:351]: Thanks for finding that, I'll submit a PR to use the syntax that will work with both versions. I prefer to keep it consistent especially in the early stages of this project and in IMO people would probably expect the credentials to be the first thing they would supply.

Comment (jborean93) [reply src/sspilib/_sec_context.py:351]: https://github.com/pyinstaller/pyinstaller-hooks-contrib/pull/712

**Pull request merged**